natsstreaming/nats_streaming.go

```diff
@@ -24,24 +24,54 @@ const (
 	OffsetNewest int64 = -1
 )
 
+// ConnectionTimeOutConfig encapsulates interval and timeout seconds fields
+// used for configuring nats streaming connection loss detection
+type ConnectionTimeOutConfig struct {
```

Comment:
This feels a bit weird as we have the same information in the source config directly, whereas in the sink config it's stored in this struct. So I would change one of these to match the other one. Dropping this extra config object would't be a breaking change and would result in smaller api surface.